cli/src/pcluster/models/cluster.py

```diff
@@ -434,6 +434,7 @@ def _validate_and_parse_config(
             LOGGER.info("Validating cluster configuration...")
             Cluster._load_additional_instance_type_data(cluster_config_dict)
             config = self._load_config(cluster_config_dict)
+            config.official_ami = self.__stack.official_ami if self.__stack else None
```

Comment:
don't you need to access `stack` property to have the ClusterStack object properly initialized? Also what happens if the call to CFN fails? do we handle the failure properly?

Reply:
>This code runs both on cluster-create and on cluster-update. The reason of this ugly if is that self.__stack doesn't exist during cluster creation. I didn't find any better way to check and I didn't want to change the code more than required for this specific change. I would appreciate any suggestion.

The problem I see here is that if no-one calls the self.stack property this code will not be executed `self.__stack = ClusterStack(AWSApi.instance().cfn.describe_stack(self.stack_name))`. Which means that accessing the AMI through `self.__stack.official_ami` relies on the assumption that at some point self.stack property has been already invoked and the ClusterStack object initialized. (if I'm not missing something :))

>However, right now I don't see how a CFN failure can affect this specific line of code: it would break before and the error handling doesn't change.

This is true if we assume that the stack object has been already initialized (which in a way is a weak assumption as you are saying). I'm ok if we feel like we want to leave it as is.

Reply:
> Which means that accessing the AMI through `self.__stack.official_ami` relies on the assumption that at some point self.stack property has been already invoked and the ClusterStack object initialized. (if I'm not missing something :))

Yes, it is correct for the update path. I believe we can reasonably rely on it and I believe it highly unlikely for the code to change so that this assumption becomes not true. Nevertheless, it is possible, so I changed the code in order to access `self.stack` property, catching and ignoring `StackNotFoundError`. I don't believe this is the best way to address the issue, but I still prefer to avoid significant code changes to solve a problem which never happened before and is unlikely to happen in the future :).